hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -1052,8 +1052,12 @@ private static List<FileSlice> getPartitionFileSlices(HoodieTableMetaClient meta
     HoodieTableFileSystemView fsView = fileSystemView.orElse(getFileSystemView(metaClient));
     Stream<FileSlice> fileSliceStream;
     if (mergeFileSlices) {
-      fileSliceStream = fsView.getLatestMergedFileSlicesBeforeOrOn(
-          partition, metaClient.getActiveTimeline().filterCompletedInstants().lastInstant().get().getTimestamp());
+      if (metaClient.getActiveTimeline().filterCompletedInstants().lastInstant().isPresent()) {
```

Comment:
We can do this more idiomatic:

```
lastInstant.map(() -> {
  getLatestMergedSlices(partition, instant)
}).getOrElse(emptyList())
```

Reply:
wondering if we need these changes or can revert it to how it was before. we have guarded it in our FileIndex and so this change is really not required.

Reply:
Deleted code doing the job is better than any other code!